### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by joelgrus**

### CODE REVIEW

**Comment by joelgrus** [comment]
this is ready for review. unfortunately, I moved all of the model components into their own directory, which broke the diffs, now every one is a deleted file / new file rather than a change, so I apologize for that.

everything seems to work right, and I tested a permalink from demo.allennlp.org in each model and they all worked.

some minor formatting may have gotten changed (a missing classname here or there), and it turns out I fixed a UX bug in the openie demo (I thought I introduced it, but it's there in the original)

File: demo/src/components/demos/MachineComprehension.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    const head = passage.slice(0, start);
+    const tail = passage.slice(start + best_span_str.length);
+
+    return (
+        <div className="model__content">
+            <OutputField label="Answer">
+            {best_span_str}
+            </OutputField>
+
+            <OutputField label="Passage Context" classes="passage">
```
Comment (aaronsarnat) [inline demo/src/components/demos/MachineComprehension.js]: Add summary prop to `OutputField`:

```jsx
            <OutputField label="Passage Context" classes="passage" summary={true}>
```

File: demo/src/components/demos/MachineComprehension.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+const Output = ({ requestData, responseData }) => {
+    const { passage } = requestData
+    const { best_span_str, passage_question_attention, question_tokens, passage_tokens } = responseData
+    const start = passage.indexOf(best_span_str);
+    const head = passage.slice(0, start);
+    const tail = passage.slice(start + best_span_str.length);
+
+    return (
+        <div className="model__content">
+            <OutputField label="Answer">
```
Comment (aaronsarnat) [inline demo/src/components/demos/MachineComprehension.js]: Add summary prop to `OutputField`:

```jsx
            <OutputField label="Answer" summary={true}>
```

File: demo/src/components/demos/WikiTables.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      )
+}
+
+
+const Output = ({ responseData }) => {
+    const { answer, logical_form, predicted_actions, linking_scores, feature_scores, similarity_scores, entities, question_tokens } = responseData
+
+    return (
+        <div className="model__content">
+          <OutputField label="Answer">
```
Comment (aaronsarnat) [inline demo/src/components/demos/WikiTables.js]: Add summary prop to `OutputField`:

```jsx
          <OutputField label="Answer" summary={true}>
```

File: demo/src/components/OutputField.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+// A labeled output field with children
+const OutputField = ({label, classes, children}) => (
+    <div className="form__field">
+        {label ? <label>{label}</label> : null}
+        <div className={`model__content__summary ${classes}`}>
+            {children}
+        </div>
+    </div>
+)
```
Comment (aaronsarnat) [inline demo/src/components/OutputField.js]: I'd make the following changes to this const:

1. Add a `summary` prop
2. Conditional nesting structure of `children` based on value of `summary`
3. Add default values to optional props `classes` and `summary`

---

Like this:

```jsx
const OutputField = ({label, classes = "", children, summary = false}) => (
    <div className="form__field">
        {label ? <label>{label}</label> : null}
        {summary ? (
          <div className={`model__content__summary ${classes}`}>
            {children}
          </div>
        ) : children}
    </div>
)
```

File: demo/ADDING_A_DEMO.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+const Output = ({ requestData, responseData }) => {
+    const { passage } = requestData
+    const { best_span_str, passage_question_attention, question_tokens, passage_tokens } = responseData
+    const start = passage.indexOf(best_span_str);
+    const head = passage.slice(0, start);
+    const tail = passage.slice(start + best_span_str.length);
+
+    return (
+        <div className="model__content">
+            <OutputField label="Answer">
```
Comment (aaronsarnat) [inline demo/ADDING_A_DEMO.md]: Add summary prop to `OutputField`:

```jsx
            <OutputField label="Answer" summary={true}>
```

File: demo/ADDING_A_DEMO.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+            {best_span_str}
+            </OutputField>
+
+            <OutputField label="Passage Context" classes="passage">
```
Comment (aaronsarnat) [inline demo/ADDING_A_DEMO.md]: Add summary prop to `OutputField`:

```jsx
            <OutputField label="Passage Context" classes="passage" summary={true}>
```

File: demo/ADDING_A_DEMO.md (hunk 5 above)
Comment (aaronsarnat) [reply demo/ADDING_A_DEMO.md]: (to keep docs up-to-date)

File: demo/ADDING_A_DEMO.md (hunk 6 above)
Comment (aaronsarnat) [reply demo/ADDING_A_DEMO.md]: (to keep docs up-to-date)

**Comment by schmmd** [comment]
@matt-gardner just an FYI--this should merge soon.

**Pull request merged**